manager/snap.go

```diff
@@ -79,6 +87,126 @@ func (snap *snap) GetProxySettings() (proxy.Settings, error) {
 	return res, nil
 }
 
+// ConfigureStoreProxyFromURL downloads an assertion list from the snap store
+// proxy located at proxyURL and sets up snapd to use the proxy and store ID
+// obtained from the retrieved assertions.
+//
+// If the local snap store proxy instance is operating in an air-gapped
+// environment, downloading the assertion list from the proxy will not be
+// possible and an appropriate error will be returned. For such use-cases, the
+// store proxy must be configured via a call to ConfigureStoreProxy instead.
+//
+// If snap also needs to use HTTP/HTTPS proxies to talk to the outside world,
+// these need to be configured separately before invoking this method via a
+// call to SetProxy.
+func (snap *Snap) ConfigureStoreProxyFromURL(proxyURL string) error {
```

Comment:
Do we want to have this function at this level? Would it be better to have something higher up the stack that sees the store-proxy-url and just reads the assertions and store id and then from then onwards interacts with *this* layer just by the store-id + assertions?

Reply:
Maybe this function becomes "get assertions from URL" ?

Reply:
That is an interesting idea. My thought was that the cloud-init bits in juju would check the configuration and call either the FromURL or the other method depending on what is available. 

I don't mind keeping just the second method but personally, I would prefer to keep the logic at a single place instead of moving this bit into the juju code base. On the other hand, we do have some snap bits in juju so it might make more sense to proceed as you suggest.

Reply:
I have removed the FromURL bits from this PR and will re-introduce it in an upcoming change-set for juju.